app/services/preserved_object_handler.rb

```diff
@@ -185,81 +152,69 @@ def create_db_objects(status, validated=false)
-        db_upd_results.concat(update_db_object(pres_copy))
-        # fine to append this result, because it's true regardless of whether the transaction succeeds
-        full_results << result_hash(ARG_VERSION_GREATER_THAN_DB_OBJECT, pres_object.class.name)
+        update_status(pres_copy, status) if status
+        update_db_object(pres_copy)
         pres_object.current_version = incoming_version
-        db_upd_results.concat(update_db_object(pres_object))
+        update_db_object(pres_object)
       else
-        # these just add result codes about object state w/o touching DB, so can append immediately to rull result list
-        full_results << result_hash(UNEXPECTED_VERSION, 'PreservedCopy')
-        full_results.concat(version_comparison_results(pres_copy, :version))
-        full_results.concat(version_comparison_results(pres_object, :current_version))
+        # add result codes about object state w/o touching DB
+        handler_results.add_result(PreservedObjectHandlerResults::UNEXPECTED_VERSION, 'PreservedCopy')
+        version_comparison_results(pres_copy, :version)
+        version_comparison_results(pres_object, :current_version)
 
-        # update_status and update_db_object both touch the db, so same circumspect handling of results from them
-        db_upd_results.concat(update_status(pres_copy, status)) if status
+        update_status(pres_copy, status) if status
         update_pc_validation_timestamps(pres_copy) if validated
-        db_upd_results.concat(update_db_object(pres_copy)) if pres_copy.changed?
+        update_db_object(pres_copy) if pres_copy.changed?
       end
     end
 
-    # ok, now we're out of the woods:  if we're here, the transaction is over.  and if it produced no results
-    # of its own, it completed and committed successfully.  so if there were no error codes produced from the
-    # transaction running and committing, return the update results, otherwise, return the transaction failure code(s).
-    if transaction_results.empty?
-      full_results.concat(db_upd_results)
-    else
-      full_results.concat(transaction_results)
-    end
+    handler_results.remove_db_updated_result unless transaction_ok
   end
 
   def confirm_version_in_catalog
-    results = []
-    confirm_results = with_active_record_transaction_and_rescue do
+    transaction_ok = with_active_record_transaction_and_rescue do
       po_db_object = PreservedObject.find_by!(druid: druid)
       pc_db_object = PreservedCopy.find_by!(preserved_object: po_db_object, endpoint: endpoint)
-      results.concat(confirm_version_on_db_object(po_db_object, :current_version))
-      results.concat(confirm_version_on_db_object(pc_db_object, :version))
+      confirm_version_on_db_object(po_db_object, :current_version)
+      confirm_version_on_db_object(pc_db_object, :version)
     end
-    results.concat(confirm_results)
+    handler_results.remove_db_updated_result unless transaction_ok
```

Comment:
adding some actual logic here dependent on transaction results ...

Reply:
slightly surprised/worried this didn't require any tests to be fixed.

Reply:
confirm_version was written early on;  i think our mental models grew over time blah blah.